### PR TITLE
added AR-reports logic

### PR DESCRIPTION
**Pull request by JungaBunga**

### CODE REVIEW

**Review by GodsonLeigh — APPROVED** [review]
approve-wiggle

File: app/routes/ap-ar-report-listing.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+                Filename: data.batch,
+                'Date Time': data.lastUpdated,
+                Event: data.status,
+                FRN: data.frn,
+                'Original Invoice Number': data.originalInvoiceNumber,
+                'Original Invoice Value': data.value,
+                'Invoice Number': data.invoiceNumber,
+                'Invoice Delta Amount': data.deltaAmount,
+                'D365 Invoice Imported': data.routedToRequestEditor,
+                'D365 Invoice Payment': data.settledValue,
```
Comment (samplackett) [inline app/routes/ap-ar-report-listing.js:78]: Can we add a step to remove this D365 Invoice Payment for the AR report? We don't get settlement data back for those so will always be null.

**Review by samplackett — CHANGES_REQUESTED** [review]
Just one small thing to tweak for the AR report to reflect the minor difference between AP and AR.

File: app/routes/ap-ar-report-listing.js [hunk 2]
```diff
@@ -74,11 +74,17 @@ function generateRoutes (reportName, reportDataUrl, reportDataKey) {
                 'Original Invoice Value': data.value,
                 'Invoice Number': data.invoiceNumber,
                 'Invoice Delta Amount': data.deltaAmount,
-                'D365 Invoice Imported': data.routedToRequestEditor,
-                'D365 Invoice Payment': data.settledValue,
-                'PH Error Status': data.phError,
-                'D365 Error Status': data.daxError
+                'D365 Invoice Imported': data.routedToRequestEditor
```
Comment (samplackett) [inline app/routes/ap-ar-report-listing.js:77]: A bit pedantic but it's probably neater to have everything in as default in mappedDate, then just say

```
if (reportName === 'ar-listing') {
  delete mappedData['D365 Invoice Payment']
}
```

Comment (samplackett) [reply app/routes/ap-ar-report-listing.js:77]: Also no semi colons! :)

**Review by samplackett — APPROVED** [review]
ainsley-lovely-nice

**Pull request merged**